src/plugin/stringSignMethods/personal-sign.ts

```diff
@@ -0,0 +1,61 @@
+import { Models } from '@open-rights-exchange/chain-js'
+import { ethers } from 'ethers'
+import { EthereumPrivateKey, PersonalSignDataInput } from '../models'
+import { convertStringToUInt8Array } from '../helpers'
+
+export async function validatePersonalSignInput(data: PersonalSignDataInput): Promise<Models.SignStringValidateResult> {
+  let result: Models.SignStringValidateResult
+
+  let message = ''
+  let valid = true
+
+  // Check that the stringToSign property exists.
+  if (!data || !data.stringToSign) {
+    message += ' stringToSign property is missing.'
+    valid = false
+  }
+
+  // Check that message is string
+  if (typeof data.stringToSign !== 'string') {
+    message += ' stringToSign property must be a string.'
+    valid = false
+  }
+
+  /* If any part of the input is not valid then let's build an example to reply with */
+  if (!valid) {
+    const fullMessage = `The data supplied to personalSign is incorrectly formatted or missing: ${message}`
+
+    const example = {
+      stringToSign: 'The message you would like to sign here',
+    }
+
+    result = {
+      valid,
+      message: fullMessage,
+      example,
+    }
+  } else {
+    result = {
+      valid: true,
+      message: '',
+      example: {},
+    }
+  }
+
+  return result
+}
+
+export async function personalSign(
+  privateKeys: EthereumPrivateKey[],
+  data: PersonalSignDataInput,
+): Promise<Models.SignStringSignResult> {
+  const privateKey = privateKeys[0]
+  const signer = new ethers.Wallet(privateKey)
+
+  const dataBytes: Uint8Array = convertStringToUInt8Array(data.stringToSign)
+  const sig = await signer.signMessage(dataBytes)
+
+  const signatureParts = ethers.utils.splitSignature(sig)
```

Comment:
do we have helpers already doing this in the plugin - is so reuse helper function